src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/AssemblyContigAlignmentSignatureClassifier.java

```diff
@@ -10,174 +9,413 @@
+        // max overlap maxFront is a tuple2 registering the index and overlap bases count
+        final List<MaxOverlapInfo> maxOverlapMap = getMaxOverlapPairs(selectedAlignments);
+        int k = 0;
+        for(Iterator<AlignmentInterval> iterator = selectedAlignments.iterator(); iterator.hasNext();) {
+            final AlignmentInterval cur = iterator.next();
+            final MaxOverlapInfo maxOverlapFrontAndRear = maxOverlapMap.get(k);
+            final int maxOverlapFront = Math.max(0, maxOverlapFrontAndRear.maxFront._2);
+            final int maxOverlapRear = Math.max(0, maxOverlapFrontAndRear.maxRear._2);
+
+            // theoretically this could be negative for an alignment whose maxFront and maxRear sums together bigger than the read span
+            // but earlier configuration scoring would make this impossible because such alignments should be filtered out already
+            // considering that it brings more penalty than value, i.e. read bases explained (even if the MQ is 60),
+            // but even if it is kept, a negative value won't hurt unless a stupid threshold value is passed in
+            final int uniqReadSpan = cur.endInAssembledContig - cur.startInAssembledContig + 1 - maxOverlapFront - maxOverlapRear;
+            if (uniqReadSpan < uniqReadLenInclusive) {
+                lowUniquenessMappings.add(cur);
+                iterator.remove();
+            }
+            ++k;
+        }
+
+        return new Tuple2<>(selectedAlignments, lowUniquenessMappings);
     }
 
     /**
-     * @param contigWithOnlyOneConfigAnd2AlnToSameChr assumed to have only two alignments mapped to the same chromosome
+     * Each alignment in a specific configuration has an entry,
+     * pointing to the alignments that comes before and after it,
+     * that overlaps maximally (i.e. no other front or rear alignments have more overlaps)
+     * with the current alignment.
      */
-    @VisibleForTesting
-    static boolean indicatesIntraChrStrandSwitchBkpts(final AlignedContig contigWithOnlyOneConfigAnd2AlnToSameChr) {
-        Utils.validateArg(isSameChromosomeMapping(contigWithOnlyOneConfigAnd2AlnToSameChr),
-                "assumption that input contig's 2 alignments map to the same chr is violated. \n" +
-                        contigWithOnlyOneConfigAnd2AlnToSameChr.toString());
-        return contigWithOnlyOneConfigAnd2AlnToSameChr.alignmentIntervals.get(0).forwardStrand
-                ^
-                contigWithOnlyOneConfigAnd2AlnToSameChr.alignmentIntervals.get(1).forwardStrand;
+    private static final class MaxOverlapInfo {
+        final Tuple2<Integer, Integer> maxFront; // 1st holds index pointing to another alignment before this, 2nd holds the count of overlapping bases
+        final Tuple2<Integer, Integer> maxRear;  // same intention as above, but for alignments after this
+
+        MaxOverlapInfo() {
+            maxFront = new Tuple2<>(-1, -1);
```

Comment:
Again, this is just for clarity and not mandatory, but what if you made `MaxOverlapInfo` have the following fields:

`AlignmentInterval maxFrontOverlapper`
`int maxFrontOverlap`
`AlignmentInterval maxRearOverlapper`
`int maxRearOverlap`

Then you don't have int indices..

Reply:
The index is actually used in the implementation that uses it to retrieve from the old list implementation. Again, I hope there's a `zip`...